.github/workflows/linux/build.bash

```diff
@@ -0,0 +1,85 @@
+set -exo pipefail
+
+export CCACHE_DIR=/icinga2/ccache
+CMAKE_OPTS=''
+
+case "$DISTRO" in
+  amazonlinux:*)
+    amazon-linux-extras install -y epel
+    yum install -y bison ccache cmake gcc-c++ flex ninja-build \
+      {libedit,mariadb,ncurses,openssl,postgresql,systemd}-devel
+
+    yum install -y bzip2 tar wget
+    wget https://boostorg.jfrog.io/artifactory/main/release/1.69.0/source/boost_1_69_0.tar.bz2
+    tar -xjf boost_1_69_0.tar.bz2
+
+    (
+      cd boost_1_69_0
+      export PATH="/usr/lib64/ccache:$PATH"
+      ./bootstrap.sh --with-libraries=context,coroutine,date_time,filesystem,iostreams,program_options,regex,system,thread
+      ./b2 link=static
+    )
```

Comment:
That's the same version as on CentOS 7, does this add any other interesting version combination? Otherwise, I'd just Amazon Linux from this job.

Reply:
Actually the same as we package for Amazon. Do you consider Amazon Linux as binary compatible as CentOS/RHEL/Rocky?

Reply:
I don't know, if Amazon claims that, but if it was, there should be no need to build Boost here in the first place, so probably not. Anyways, my goal for GitHub Actions would be to cover a useful set of compiler/library versions, rather than trying to match our package builds as close as possible. For this, we have out GitLab CI pipelines.

Reply:
## gcc-c++

* amazonlinux:2: 7.3.1-15.amzn2
* centos:7: 4.8.5-44.el7

Enough for me to also build it.

Reply:
> Anyways, my goal for GitHub Actions would be to cover a useful set of compiler/library versions, rather than trying to match our package builds as close as possible. For this, we have out GitLab CI pipelines.

The pipeline for Amazon Linux 2 does not work at the moment. And I wouldn't put effort into fixing this, also saves us from having to build Boost each time.

Reply:
I've already fixed this.